Controller/CRUDController.php

```diff
@@ -1298,6 +1308,21 @@ protected function getCsrfToken($intention)
         return false;
     }
 
+
+    /**
+     * This method can be overloaded in your custom CRUD controller.
+     * It's called from createAction and editAction.
+     *
+     * @param Form $form
+     * @param mixed   $object
+     *
+     * @return Response|null
+     */
+    protected function preFormCheck(Form $form, $object)
+    {
+
```

Comment:
see https://github.com/sonata-project/SonataAdminBundle/blob/3.x/Controller/CRUDController.php#L247
and https://github.com/sonata-project/SonataAdminBundle/blob/3.x/Controller/CRUDController.php#L1323

i formatted the method following this example